core/src/outputs.rs

```diff
@@ -142,3 +142,34 @@ fn are_valid_elements(outputs: &[u64]) -> bool {
     }
     true
 }
+
+impl Serializable for StackOutputs {
+    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
+        self.stack.write_into(target)?;
+        self.overflow_addrs.write_into(target)?;
+        Ok(())
+    }
+}
+
+impl Deserializable for StackOutputs {
+    fn read_from(bytes: &mut ByteReader) -> Result<Self, SerializationError> {
+        let stack = Vec::read_from(bytes)?;
+        let overflow_addrs = Vec::read_from(bytes)?;
+        Ok(Self {
+            stack,
+            overflow_addrs,
+        })
+    }
+}
+
+impl winter_utils::Serializable for StackOutputs {
+    fn write_into<W: winter_utils::ByteWriter>(&self, target: &mut W) {
+        // stack
+        target.write_u64(self.stack.len() as u64);
+        self.stack.iter().copied().for_each(|v| target.write_u64(v));
+
+        // overflow addrs
+        target.write_u64(self.overflow_addrs.len() as u64);
+        self.overflow_addrs.iter().copied().for_each(|v| target.write_u64(v));
+    }
```

Comment:
Similar comment as before. I think this can be problematic, and I'd prefer to get rid of non-winterfell serialization from here until we find a better solution.

Reply:
Ack! Resolving this discussion to keep the topic centralized on the previous one